crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -536,8 +552,97 @@ async fn reduce_config(
+    let client = state.redis.next_connected();
+    let config = client.get::<String, String>(config_key).await;
+    let metrics = client.take_latency_metrics();
+    let network_metrics = client.take_network_latency_metrics();
+    log::trace!(
+        "Network metrics for config fetch in milliseconds :: max: {}, min: {}, avg: {}; Latency metrics :: max: {}, min: {}, avg: {}",
+        network_metrics.max,
+        network_metrics.min,
+        network_metrics.avg,
+        metrics.max,
+        metrics.min,
+        metrics.avg
+    );
+    match config {
+        Ok(config) => {
+            let mut response = HttpResponse::Ok();
+            if let Ok(max_created_at) =
+                client.get::<String, String>(max_created_key).await
+            {
+                let metrics = client.take_latency_metrics();
+                let network_metrics = client.take_network_latency_metrics();
+                log::trace!(
+                    "Network metrics max-created-by fetch in milliseconds :: max: {}, min: {}, avg: {}; Latency metrics :: max: {}, min: {}, avg: {}",
+                    network_metrics.max,
+                    network_metrics.min,
+                    network_metrics.avg,
+                    metrics.max,
+                    metrics.min,
+                    metrics.avg
+                );
+                response
+                    .insert_header((AppHeader::LastModified.to_string(), max_created_at));
+            }
+            if let Ok(audit_id) = client.get::<String, String>(audit_id_key).await {
+                let metrics = client.take_latency_metrics();
+                let network_metrics = client.take_network_latency_metrics();
+                log::trace!(
+                    "Network metrics for audit ID in milliseconds :: max: {}, min: {}, avg: {}; Latency metrics :: max: {}, min: {}, avg: {}",
+                    network_metrics.max,
+                    network_metrics.min,
+                    network_metrics.avg,
+                    metrics.max,
+                    metrics.min,
+                    metrics.avg
+                );
+                response.insert_header((AppHeader::XAuditId.to_string(), audit_id));
+            }
+            if let Ok(config_version) =
+                client.get::<Option<i64>, String>(config_version_key).await
+            {
+                let metrics = client.take_latency_metrics();
+                let network_metrics = client.take_network_latency_metrics();
+                log::trace!(
+                    "Network metrics for version ID in milliseconds :: max: {}, min: {}, avg: {}; Latency metrics :: max: {}, min: {}, avg: {}",
+                    network_metrics.max,
+                    network_metrics.min,
+                    network_metrics.avg,
+                    metrics.max,
+                    metrics.min,
+                    metrics.avg
+                );
+                add_config_version_to_header(&config_version, &mut response);
```

Comment:
:laughing: this helper function is just a wrapper over 1 line, we will have to clean all this up later.

Reply:
Yeah agreed